### DISCUSSION TITLE
Destroy all decryption keys from a room, onec all users leave it.

### DISCUSSION THREAD

**Discussion by Atom-Dragon** [element-hq/element-meta#655] [Ideas]
### Your use case

It would be nice if all decryption keys are deleted from a room when all members leave it.

For example say you're homeserver is matrix.org which is using EMS, if all users were to leave the room the contains in that room would still exist but it would be un-joinable. Since all users are unable to join back it would be good from a privacy and security stand point to delete the keys as all the users can't really access that left room anymore.

I was told on EMS homeservers the function "purge room on all members leave" is not enabled by default as it's impossible to do on EMS, hence deleting all the decryption of a forgotten room would be a great feature to have and give users extra protection in case of a future beach.

### Have you considered any alternatives?

An Alternative to this would be to have an automatic garbage collection of empty rooms, as having the empty rooms would take unnecessary space on the homeserver especially if the rooms contains media and files. so once in a while a function is ran to clear out any rooms where there is no users.

**Comment by t3chguy** [comment]
Users can look back at history of rooms they have left, so they'd need to keep keys to decrypt that history

**Reply by Atom-Dragon** [reply]
Hmmm I know that's the case of rooms where there are users still in them, then yes you would still be able to see the history of the conversation till the point you leave the room. However I am unable to see the history of rooms where all members have left it. When I click on the room history it's just show invalid. I am sure the room is not deleted because the "purge room on leave" is not enabled on the matrix homeserver by default. Unless I am missing something and the rooms are eventually deleted and that's why I am unable to see the history.

**Reply by t3chguy** [reply]
> When I click on the room history it's just show invalid

Then this is a client bug, the history is still available to you until the point you opt to manually "Forget" the room

**Reply by Atom-Dragon** [reply]
Actually, my bad it's not a client bug I can see the room history however, if a room is set to private and you leave the room it will show on the history panel with the message" in order to see the history of the room you will need to join it" now in this case since there is no users in the room, I can't be reinvited and pressing the join room button dose not work so you can't really see the history even after you leave. Would this be a good enough reason to delete the keys if the condition of the room is private and all users have left the room.

**Reply by t3chguy** [reply]
> in order to see the history of the room you will need to join it

Again, client bug, the API still makes the data available via historical rooms
